### PR TITLE
Check Kubernetes Version, use correct API, services per Release and Fix logging

### PR DESCRIPTION
**Pull request by slowr**
### Description of PR

New kubernetes version deprecated some APIs. To have backwards compatibility I created a check that uses new API on new versions.

What component(s) does this PR affect?

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [x] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

### Related Issue

Resolves #

### Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: artemis-chart/templates/backend-deployment.yaml [hunk 1]
```diff
@@ -26,7 +28,7 @@ spec:
         - mountPath: /pvc
           name: backend-pvc
           subPath: configs
-        command: ['sh', '-c', 'cp -n -u /configmaps/config.yaml /configmaps/logging.yaml /configmaps/services.conf /pvc/']
+        command: ['sh', '-c', 'false | cp -i /configmaps/config.yaml /configmaps/logging.yaml /configmaps/services.conf /pvc/ 2> /dev/null']
```
Comment (vkotronis) [inline artemis-chart/templates/backend-deployment.yaml]: What is this false about?

File: artemis-chart/templates/monitor-deployment.yaml [hunk 2]
```diff
@@ -26,7 +28,7 @@ spec:
         - mountPath: /pvc
           name: monitor-pvc
           subPath: configs
-        command: ['sh', '-c', 'cp -n -u /configmaps/logging.yaml /configmaps/mon-services.conf /pvc/']
+        command: ['sh', '-c', 'false | cp -i /configmaps/logging.yaml /configmaps/mon-services.conf /pvc/ 2> /dev/null']
```
Comment (vkotronis) [inline artemis-chart/templates/monitor-deployment.yaml]: shouldn't also the changes regarding
```
        {{- range $key, $value := .Values.nodeSelector }}
        {{ $key }}: {{ $value }}
        {{- end }}
      {{ toYaml . | nindent 8 }}
      {{- end }}
```
applied here as in the other deployments?

File: artemis-chart/templates/nginx-deployment.yaml [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+      app: {{ .Release.Name }}-nginx
   replicas: {{ .Values.replicaCount | default 1 }}
   strategy:
     type: Recreate
   template:
     metadata:
       creationTimestamp: null
       labels:
-        io.kompose.service: nginx
+        app: {{ .Release.Name }}-nginx
```
Comment (vkotronis) [inline artemis-chart/templates/nginx-deployment.yaml]: shouldn't also the changes regarding
```
        {{- range $key, $value := .Values.nodeSelector }}
        {{ $key }}: {{ $value }}
        {{- end }}
      {{ toYaml . | nindent 8 }}
      {{- end }}
```
applied here as in the other deployments?

File: artemis-chart/templates/nginx-service.yaml [hunk 4]
```diff
@@ -2,9 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  labels:
-    io.kompose.service: nginx
-  name: {{ .Release.Name }}-nginx
+  name: nginx
```
Comment (vkotronis) [inline artemis-chart/templates/nginx-service.yaml]: why not release name here?

File: artemis-chart/templates/postgres-deployment.yaml [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
+      app: {{ .Release.Name }}-postgres
   replicas: {{ .Values.replicaCount | default 1 }}
   strategy:
     type: Recreate
   template:
     metadata:
       creationTimestamp: null
       labels:
-        io.kompose.service: postgres
+        app: {{ .Release.Name }}-postgres
```
Comment (vkotronis) [inline artemis-chart/templates/postgres-deployment.yaml]: shouldn't also the changes regarding
```
        {{- range $key, $value := .Values.nodeSelector }}
        {{ $key }}: {{ $value }}
        {{- end }}
      {{ toYaml . | nindent 8 }}
      {{- end }}
```
applied here as in the other deployments?

File: artemis-chart/templates/pvc.yaml [hunk 6]
```diff
@@ -2,8 +2,6 @@ apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
   name: {{ .Release.Name }}-postgres-pvc
-  labels:
-    app: {{ .Release.Name }}
```
Comment (vkotronis) [inline artemis-chart/templates/pvc.yaml]: why are those deleted?

File: artemis-chart/templates/backend-deployment.yaml (hunk 1 above)
Comment (slowr) [reply artemis-chart/templates/backend-deployment.yaml]: it asks if you want to replace the files and it answer no :P

File: artemis-chart/templates/pvc.yaml (hunk 6 above)
Comment (slowr) [reply artemis-chart/templates/pvc.yaml]: not needed

File: frontend/webapp/configs/logging.yaml [hunk 7]
```diff
@@ -36,7 +36,7 @@ handlers:
     #     address: [localhost, /dev/log]
 
 loggers:
-    webapp_logger:
+    artemis_logger:
```
Comment (vkotronis) [inline frontend/webapp/configs/logging.yaml]: reminder: people will need to change this in their local_configs, otherwise things will break on the frontend

File: frontend/webapp/configs/config.py [hunk 8]
```diff
@@ -74,7 +74,7 @@ def configure_app(app):
     app.config.from_object("webapp.configs.config.BaseConfig")
     app.config.from_pyfile("/etc/artemis/webapp.cfg", silent=False)
 
-    app.artemis_logger = logging.getLogger("webapp_logger")
+    app.artemis_logger = logging.getLogger("artemis_logger")
```
Comment (vkotronis) [inline frontend/webapp/configs/config.py]: also needs to be changed on local_configs. Are you sure we need this change? Why can't we keep it as webapp_logger and avoid all potential issues with users forgetting to change this locally (on non-git-controlled files)?

Comment (slowr) [reply frontend/webapp/configs/config.py]: I decided to do it this way to keep consistency. Other logging yaml files are artemis_logger and also the function that actually prints is artemis_logger as well

Comment (slowr) [reply frontend/webapp/configs/config.py]: but I can keep it the old way if we want to have this syntax

Comment (vkotronis) [reply frontend/webapp/configs/config.py]: It is probably OK, but put a reminder somewhere to warn people to change their logging.yaml files for the frontend (and config.py), if they upgrade to the new version.

**Comment by vkotronis** [comment]
@slowr reminder:
1) update also the CHANGELOG with your exact changes
2) note that we will need to inform all deployment users to update their frontend config and config.py to use the artemis_logger instead of the webapp_logger, correct? Can this be done in an easy-to-migrate way (no breaks in logging)?

**Comment by vkotronis** [comment]
@slowr I saw the changes on CHANGELOG, seem to cover the backwards-incompatible changes, thanks! Ready to merge on my side, just do a final double-check with Alistair and is ready to go.
